src/components/organisms/EntityList/EntityCreationModal/TestCreationForm/TestCreationForm.tsx

```diff
@@ -0,0 +1,183 @@
+  const remappedCustomTestSources = remapTestSources(testSources);
+
+  const [localLabels, setLocalLabels] = useState<readonly Option[]>([]);
+
+  const [addTest, {isLoading}] = useAddTestMutation();
+
+  const onSave = (values: any) => {
+    const {testSource, testType} = values;
+
+    const requestBody = {
+      name: values.name,
+      type: testType,
+      labels: decomposeLabels(localLabels),
+      content: getSourcePayload(values, testSources),
+      ...getCustomSourceField(testSource),
+    };
+
+    return addTest(requestBody)
+      .then(res => {
+        onSuccess(res);
+      })
+      .catch(err => {
+        onFail(err);
+      });
+  };
+
+  return (
+    <Form form={form} layout="vertical" name="test-creation" onFinish={onSave} style={{flex: 1}} labelAlign="right">
+      <StyledFormSpace size={24} direction="vertical">
+        <Text className="regular big">Test details</Text>
+        <FormItem
+          name="name"
+          label="Name"
+          rules={[required, k8sResourceNamePattern, k8sResourceNameMaxLength]}
+          required
+        >
+          <Input placeholder="e.g.: my-test" />
+        </FormItem>
+        <FormItem name="testType" label="Type" rules={[required]} required>
+          <Select placeholder="Select from available executors..." showSearch options={remappedExecutors} allowClear />
+        </FormItem>
+        <FormItem
+          noStyle
+          shouldUpdate={(prevValues, currentValues) => prevValues.testSource !== currentValues.testSource}
+        >
+          {({getFieldValue}) => {
+            const selectedExecutor = executors.find((executor: Executor) =>
+              executor.executor?.types?.includes(getFieldValue('testType'))
+            );
+
+            if (!selectedExecutor) {
+              return null;
+            }
+
+            const options = [
+              ...remappedCustomTestSources,
+              ...testSourceBaseOptions.filter(
+                option =>
+                  selectedExecutor.executor?.contentTypes?.includes(String(option.value)) ||
+                  !selectedExecutor.executor?.contentTypes?.length
+              ),
+            ];
+
+            return (
+              <FormItem rules={[required]} label="Source" name="testSource" required>
+                <Select placeholder="Select a source..." options={options} showSearch allowClear />
+              </FormItem>
+            );
+          }}
+        </FormItem>
+        <FormItem
+          noStyle
+          shouldUpdate={(prevValues, currentValues) => prevValues.testSource !== currentValues.testSource}
+        >
+          {({getFieldValue}) => {
+            const testSourceValue = getSourceFieldValue(getFieldValue);
+
+            const selectedExecutor = executors.find((executor: any) =>
+              executor.executor?.types?.includes(getFieldValue('testType'))
+            );
+            const executorType = selectedExecutor?.executor.meta?.iconURI;
```

Comment:
If the `selectedExecutor` would be on top, this could be extracted as well

Reply:
Agree, will move it